lms/djangoapps/branding_by_referer/configuration_by_referer.py

```diff
@@ -0,0 +1,161 @@
+"""
+Overwites some site configuration according to the referer
+"""
+import copy
+import json
+
+from django.utils.deprecation import MiddlewareMixin
+from django.utils.six import iteritems
+from django.utils.six.moves.urllib.parse import urlparse
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_api.models import UserPreference
+
+
+class SetConfigurationByReferer(MiddlewareMixin):
+    """
+    This middleware checks if the current logged-in user has a marketing site referer preference.
+    If so, it overwrites some values of the site configuration based on the referer.
+    It depends on the SetBrandingByReferer middleware to apply the configuration overwrites, as that
+    middleware sets the referer preference for the current logged-in user. Therefore this middleware
+    must be located below the SetBrandingByReferer one in the MIDDLEWARE_CLASSES setting
```

Comment:
can you add this into the README as well

Reply:
Sure